adaptdl/adaptdl/torch/data.py

```diff
@@ -147,6 +147,16 @@ def __init__(self, batch_size=1):
         self.batch_size = batch_size
         self.future_exit = None
 
+        # True before the first time sync_local_bsz.
+        self.init_time = True
+
+        # initialize a record of current local bs, if the sppedup 
+        # is not significant then we will keep the previous batch size
+        self.cur_local_bsz = 0
```

Comment:
Don't think you need to introduce these two new fields. There's the existing `current_local_bsz` which you can just initialize to `None` here instead.